src/thread_local_heap.h

```diff
@@ -108,7 +108,8 @@ class ThreadLocalHeap {
     return _global->pageAlignedAlloc(pageAlignment, pageCount);
   }
 
-  inline void *ATTRIBUTE_ALWAYS_INLINE realloc(void *oldPtr, size_t newSize) {
+  inline void *ATTRIBUTE_ALWAYS_INLINE ATTRIBUTE_ALLOC_SIZE(3)
```

Comment:
@devnexen @emeryberger shouldn't this be `ATTRIBUTE_ALLOC_SIZE(2)`? (according to https://gcc.gnu.org/onlinedocs/gcc-4.7.2/gcc/Function-Attributes.html )